src/dex/maverick/maverick-events.test.ts

```diff
@@ -0,0 +1,126 @@
+import dotenv from 'dotenv';
+dotenv.config();
+
+import { MaverickEventPool } from './maverick';
+import { MaverickConfig } from './config';
+import { Network } from '../../constants';
+import { DummyDexHelper } from '../../dex-helper/index';
+import { testEventSubscriber } from '../../../tests/utils-events';
+import { MaverickPoolState, SubgraphPoolBase } from './types';
+import { Tokens, Holders } from '../../../tests/constants-e2e';
+import axios from 'axios';
+
+jest.setTimeout(50 * 1000);
+const dexKey = 'Maverick';
+const network = Network.POLYGON;
+const config = MaverickConfig[dexKey][network];
+
+async function fetchPoolState(
+  maverickPool: MaverickEventPool,
+  blockNumber: number,
+): Promise<MaverickPoolState> {
+  return maverickPool.generateState(blockNumber);
+}
+
+async function getSubgraphPool(
+  address: string,
+  blockNumber: number,
+): Promise<SubgraphPoolBase> {
+  address = address.toLowerCase();
+
+  const query = `query ($blockNumber: Int, $address: Bytes!) {
+    pools(block: { number: $blockNumber }, where: {id: $address}) {
+        id  
+        fee
+        w
+        h
+        k
+        paramChoice
+        twauLookback
+        uShiftMultiplier
+        maxSpreadFee
+        spreadFeeMultiplier
+        protocolFeeRatio
+        epsilon
+        quoteBalance
+        baseBalance
+        base {
+            id
+            decimals
+            symbol
+        }
+        quote {
+            id
+            decimals
+            symbol
+        }
+    }
+  }`;
+
+  const variables = {
+    blockNumber,
+    address,
+  };
+
+  const data = await axios.post(
+    config.subgraphURL,
+    { query, variables },
+    { timeout: 5000 },
+  );
+  return data.data.data.pools[0];
+}
+
+describe('Maverick Event', function () {
+  const blockNumbers: { [eventName: string]: number[] } = {
+    AddLiquidity: [29299844],
+    Swap: [29300959, 29301354, 29301378],
```

Comment:
Do you have more events? It would be good to include `RemoveLiquidity` too. As a rule of thumb we check at least 5 events if possible for each event